File: spring/src/main/java/com/example/eurasia/entity/Data/ExcelReportSettingData.java
package com.example.eurasia.entity.Data;

public class ExcelReportSettingData {
    // query conditions
    private QueryCondition[] queryConditions = null;
    // excel report types
    private String[] excelReportTypes = null;

    public ExcelReportSettingData() {
    }

    public ExcelReportSettingData(QueryCondition[] queryConditions, String[] excelReportTypes) {
        this.queryConditions = queryConditions;
        this.excelReportTypes = excelReportTypes;
    }

    // set/get query conditions
    public void setQueryConditions(QueryCondition[] queryConditions) {
        this.queryConditions = queryConditions;
    }
    public QueryCondition[] getQueryConditions() {
        return this.queryConditions;
    }
    // set/get excel report types
    public void setExcelReportTypes(String[] excelReportTypes) {
        this.excelReportTypes = excelReportTypes;
    }
    public String[] getExcelReportTypes() {
        return this.excelReportTypes;
    }
}

File: angular/src/app/data-search/entities/excel-report-setting-data.ts
import { QueryCondition } from './query-condition';

export class ExcelReportSettingData {
  // query conditions
  private queryConditions: QueryCondition[] = null;
  // excel report types
  private excelReportTypes: string[] = null;
  // set/get/query conditions
  public setQueryConditions(queryConditions: QueryCondition[]): void {
    this.queryConditions = queryConditions;
  }
  public getQueryConditions(): QueryCondition[] {
    return this.queryConditions;
  }
  // set/get excel report types
  public setExcxelReportTypes(excelReportTypes: string[]): void {
    this.excelReportTypes = excelReportTypes;
  }
  public getExcelReportTypes(): string[] {
    return this.excelReportTypes;
  }
}
